syntax = "proto3";
package heron.proto.system;

option java_package = "com.twitter.heron.proto.system";
option java_outer_classname = "ExecutionEnvironment";

message HeronReleaseState {
  string release_username = 1;
  string release_tag = 2;
  // This is the user facing release version
  string release_version = 3;
}

message ExecutionState {
  string topology_name = 1;
  string topology_id = 2;
  // The time the topology was submitted
  // Unix time
  int64 submission_time = 3;
  // The username who launched the topology
  string submission_user = 4;

  string cluster = 5;
  string environ = 6;
  string role = 7;
  HeronReleaseState release_state = 8;
}
